core/src/main/java/org/infinispan/util/concurrent/CompletableFutures.java

```diff
@@ -105,6 +110,29 @@ public static boolean await(CompletableFuture<?> future, long time, TimeUnit uni
       }
    }
 
+   /**
+    * Same as {@link #await(CompletableFuture, long, TimeUnit)} but wraps checked exceptions in {@link CacheException}
+    *
+    * @param future the {@link CompletableFuture} to test.
+    * @param time   the timeout.
+    * @param unit   the timeout unit.
+    * @return {@code true} if completed, {@code false} if timed out.
+    * @throws NullPointerException if {@code future} or {@code unit} is {@code null}.
+    */
+   public static boolean uncheckedAwait(CompletableFuture<?> future, long time, TimeUnit unit) {
+      try {
+         requireNonNull(future, "Completable Future must be non-null.").get(time, requireNonNull(unit, "Time Unit must be non-null"));
+         return true;
+      } catch (ExecutionException e) {
+         return true;
```

Comment:
I would say this should wrap the underlying exception in a new `CacheException`.

To be honest now that I see `await` does the same thing, I am not a fan. It basically requires the user to invoke `get` on the future after it returns to properly see the exception.

Reply:
My use case is to wait for it to complete without caring if the CF is completed successfully or not.